compiler/src/iree/compiler/Codegen/Common/test/combine_layout_transformation.mlir

```diff
@@ -272,3 +272,37 @@ func.func @propagate_relayout_ops(%source : tensor<?x?x128x128xf32>,
 //  CHECK-SAME:     outs(%[[INIT]] : tensor<?x?x128x128xf16>)
 //       CHECK:   %[[MAP_SCATTER:.+]] = iree_linalg_ext.map_scatter %[[COMPUTE_OP]]
 //       CHECK:   iree_codegen.store_to_buffer %[[MAP_SCATTER]]
+
+// -----
+
+func.func @map_scatter_op(%2 : tensor<196608x35xbf16>, %3 : tensor<196608x35xbf16>, %9 : tensor<8x16x1x16xbf16>, %10 : tensor<8x16x1x16xbf16>) -> (tensor<196608x35xbf16>, tensor<196608x35xbf16>){
+  %6:2 = scf.forall (%arg0, %arg1) = (0, 0) to (196608, 35) step (128, 16) shared_outs(%arg2 = %2, %arg3 = %3) -> (tensor<196608x35xbf16>, tensor<196608x35xbf16>) {
+    %collapsed = tensor.collapse_shape %9 [[0, 1], [2, 3]] : tensor<8x16x1x16xbf16> into tensor<128x16xbf16>
+    %collapsed_1 = tensor.collapse_shape %10 [[0, 1], [2, 3]] : tensor<8x16x1x16xbf16> into tensor<128x16xbf16>
+    scf.forall.in_parallel {
+      tensor.parallel_insert_slice %collapsed into %arg2[%arg0, %arg1] [128, 16] [1, 1] : tensor<128x16xbf16> into tensor<196608x35xbf16>
+      tensor.parallel_insert_slice %collapsed_1 into %arg3[%arg0, %arg1] [128, 16] [1, 1] : tensor<128x16xbf16> into tensor<196608x35xbf16>
+    }
+  } {mapping = [#iree_codegen.workgroup_mapping<y>, #iree_codegen.workgroup_mapping<x>]}
+  return %6#0, %6#1  : tensor<196608x35xbf16>, tensor<196608x35xbf16>
+}
+
+// CHECK-LABEL: @map_scatter_op
+//       CHECK:   %[[MAP_SCATTER1:.+]] = iree_linalg_ext.map_scatter
+//       CHECK:   %[[MAP_SCATTER2:.+]] = iree_linalg_ext.map_scatter
+//       CHECK:        } {mapping = [#iree_codegen.workgroup_mapping<y>, #iree_codegen.workgroup_mapping<x>]}
+
+// -----
+
+func.func @no_map_scatter_op(%2 : tensor<196608x35xbf16>, %9 : tensor<8x16x1x16xbf16>) -> tensor<196608x35xbf16>{
+  %6 = scf.forall (%arg0, %arg1) = (0, 0) to (196608, 35) step (128, 16) shared_outs(%arg2 = %2) -> (tensor<196608x35xbf16>) {
+    %collapsed = tensor.collapse_shape %9 [[0, 1], [2, 3]] : tensor<8x16x1x16xbf16> into tensor<128x16xbf16>
+    scf.forall.in_parallel {
+      tensor.parallel_insert_slice %collapsed into %arg2[%arg0, %arg1] [128, 16] [1, 1] : tensor<128x16xbf16> into tensor<196608x35xbf16>
+    }
+  } {mapping = [#gpu.thread<linear_dim_1>, #gpu.thread<linear_dim_0>]}
+  return %6 : tensor<196608x35xbf16>
+}
+// CHECK-LABEL: @no_map_scatter_op
+//   CHECK-NOT:   %[[MAP_SCATTER:.+]] = iree_linalg_ext.map_scatter
+//   CHECK-NOT:        } {mapping = [#iree_codegen.workgroup_mapping<y>, #iree_codegen.workgroup_mapping<x>]}
```

Comment:
I dont think you need to check the absence of workgroup mapping so you can remove this check.